### PR TITLE
[GH-50] Added slash command autocomplete functionality

### PR DESCRIPTION
**Pull request by pazdev**
#### Summary
Added slash command autocomplete functionality, based on 5.24 MM server 

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-plugin-welcomebot/issues/50

### CODE REVIEW

File: server/command.go [hunk 1]
```diff
@@ -246,3 +247,25 @@ func (p *Plugin) ExecuteCommand(_ *plugin.Context, args *model.CommandArgs) (*mo
 	p.postCommandResponse(args, "Unknown action %v", action)
 	return &model.CommandResponse{}, nil
 }
+
+func getAutocompleteData() *model.AutocompleteData {
+	welcomebot := model.NewAutocompleteData("welcomebot", "[command]",
+		"Available commands: preview, help, list, set_channel_welcome, get_channel_welcome, delete_channel_welcome")
+
+	preview := model.NewAutocompleteData("preview", "[team-name]", "Preview the welcome message for the given team name")
```
Comment (hanzei) [inline server/command.go:255]: `team-name` should be a text argument similar to https://github.com/mattermost/mattermost-plugin-github/blob/6957b7f8e97a96030982b1c527713e486e41afbd/server/plugin/command.go#L541

File: server/command.go [hunk 2]
```diff
@@ -246,3 +247,25 @@ func (p *Plugin) ExecuteCommand(_ *plugin.Context, args *model.CommandArgs) (*mo
...
+	welcomebot.AddCommand(preview)
+
+	list := model.NewAutocompleteData("list", "", "Lists team welcome messages")
+	welcomebot.AddCommand(list)
+
+	setChannelWelcome := model.NewAutocompleteData("set_channel_welcome", "[welcome-message]", "Set the welcome message for the channel")
```
Comment (hanzei) [inline server/command.go:261]: Same here for `welcome-message`

**Review by hanzei — CHANGES_REQUESTED** [review]
Congratulations to your first PR :tada: 

A few comments below

File: go.sum [hunk 3]
```diff
@@ -18,10 +18,14 @@ github.com/DataDog/datadog-go v3.2.0+incompatible/go.mod h1:LButxg5PwREeZtORoXG3
 github.com/Masterminds/glide v0.13.2/go.mod h1:STyF5vcenH/rUqTEv+/hBXlSTo7KYwg2oc2f4tzPWic=
```
[the remark this answers is not in the document]
Comment (pazdev) [reply go.sum:18]: Committed at https://github.com/mattermost/mattermost-plugin-welcomebot/pull/69/commits/372ed8ccc563b8b47ebffc6fdb4a5c868dd7c0a3

File: server/command.go (hunk 1 above)
Comment (pazdev) [reply server/command.go:255]: Committed at https://github.com/mattermost/mattermost-plugin-welcomebot/pull/69/commits/ffeadbc340f17b1c44c62fa9f7c586b85aadb859

File: server/command.go (hunk 2 above)
Comment (pazdev) [reply server/command.go:262]: Committed at https://github.com/mattermost/mattermost-plugin-welcomebot/pull/69/commits/ffeadbc340f17b1c44c62fa9f7c586b85aadb859

**Review by hanzei — APPROVED** [review]
LGTM :+1:

File: server/command.go [hunk 4]
```diff
@@ -246,3 +247,27 @@ func (p *Plugin) ExecuteCommand(_ *plugin.Context, args *model.CommandArgs) (*mo
 	p.postCommandResponse(args, "Unknown action %v", action)
 	return &model.CommandResponse{}, nil
 }
+
+func getAutocompleteData() *model.AutocompleteData {
+	welcomebot := model.NewAutocompleteData("welcomebot", "[command]",
+		"Available commands: preview, help, list, set_channel_welcome, get_channel_welcome, delete_channel_welcome")
+
+	preview := model.NewAutocompleteData("preview", "[team-name]", "Preview the welcome message for the given team name")
+	preview.AddTextArgument("Team Name to preview welcome message", "[team-name]", "")
```
Comment (jfrerich) [inline server/command.go:256]: One small nitpick and this PR looks good!

```suggestion
	preview.AddTextArgument("Team name to preview welcome message", "[team-name]", "")
```

**Review by jfrerich — APPROVED** [review]
One small nit.

File: server/command.go (hunk 4 above)
Comment (pazdev) [reply server/command.go]: Updated with changes https://github.com/mattermost/mattermost-plugin-welcomebot/pull/69/commits/c030af75579b6538197badf8ff65273bcf169de8

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Autocomplete implemented and works as expected
- All available commands are included and functional
- Min version updated to ensure auto-complete will be available


LGTM!
Thank @pazdev  fir this enhancement! Much appreciated :tada:

I did find an unrelated issue and created https://github.com/mattermost/mattermost-plugin-welcomebot/issues/72 to address it.

**Pull request merged**